test/app/App/Panel/AddToken/index.test.js

```diff
@@ -7,88 +7,151 @@ import store from '../../../../../main/store'
+      return Promise.resolve(tokenData)
+    }
 
-  const tokenChainName = getByRole('option', { selected: true }).textContent
-  expect(tokenChainName).toEqual('Mainnet')
+    return Promise.resolve({})
+  })
 })
 
-it('should generate the expected HTML', async () => {
-  const { asFragment } = render(
-    <AddToken activeChains={[{ id: 1, name: 'Mainnet', connection: { primary: { connected: true } } }, { id: 137, name: 'Polygon', connection: { primary: { connected: true } } }]} />
-  )
+describe('selecting token chain', () => {
+  it('should display the expected chain IDs', async () => {
+    const { getAllByRole } = render(
+      <AddToken activeChains={[{ id: 1, name: 'Mainnet', connection: { primary: { connected: true } } }, { id: 137, name: 'Polygon', connection: { primary: { connected: true } } }]} />
+    )
 
-  expect(asFragment()).toMatchSnapshot()
-})
+    const tokenChainNames = getAllByRole('button').map((el) => el.textContent)
+    expect(tokenChainNames).toEqual(['Mainnet', 'Polygon'])
+  })
 
-describe('token metadata lookup', () => {
-  beforeEach(() => {
-    link.invoke.mockImplementation((channel, contractAddress, chainId) => {
-      const tokenData = chainId === 1 ? { name: 'Frame Test', symbol: 'FRT', decimals: '18' } : { name: 'Frame Test on Polygon', symbol: 'mFRT', decimals: '18' }
-      return Promise.resolve(tokenData)
-    })
+  it('should generate the expected HTML', async () => {
+    const { asFragment } = render(
+      <AddToken activeChains={[{ id: 1, name: 'Mainnet', connection: { primary: { connected: true } } }, { id: 137, name: 'Polygon', connection: { primary: { connected: true } } }]} />
+    )
+
+    expect(asFragment()).toMatchSnapshot()
   })
+})
+
+describe('entering token contract address', () => {
+  it('should prompt for the token contract address', async () => {
+    const { getByText, findByLabelText, findByText } = render(
+      <AddToken activeChains={[{ id: 1, name: 'Mainnet', connection: { primary: { connected: true } } }, { id: 137, name: 'Polygon', connection: { primary: { connected: true } } }]} />
+    )
+    await user.click(getByText('Polygon'))
+    
+    await findByLabelText(`What is the token's contract address?`, { selector: 'input' })
+    await findByText('on Polygon')
+  }, 1000)
+
+  it('should generate the expected HTML', async () => {
+    const { asFragment, getByText } = render(
+      <AddToken activeChains={[{ id: 1, name: 'Mainnet', connection: { primary: { connected: true } } }, { id: 137, name: 'Polygon', connection: { primary: { connected: true } } }]} />
+    )
+    await user.click(getByText('Polygon'))
+
+    await waitFor(() => expect(asFragment()).toMatchSnapshot())
+  }, 1000)
+})
 
-  it('should perform a lookup on a contract address and display the expected token data', async () => {
-    const { getByLabelText, getByRole } = render(
+describe('retrieving token metadata - successful lookup', () => {
```

Comment:
probably should break these into a `describe` and individual tests like "successful lookup", "unsuccessful lookup", etc